Makefile

```diff
@@ -58,13 +58,11 @@ test-compile-changed: os-autoinst/
 
 .PHONY: test-yaml-valid
 test-yaml-valid:
-	@# Get list of changed yaml files. We only want to lint changed files for
-	@# now because yamllint complains about a lot of existing files
-	$(eval CHANGED_YAMLS=$(shell sh -c "git --no-pager diff --diff-filter=dr --name-only origin/master | grep '\(schedule\|test_data\)/.*\.y.\?ml$$'"))
-	if test -n "$(CHANGED_YAMLS)"; then \
-      export PERL5LIB=${PERL5LIB_} ; tools/test_yaml_valid $(CHANGED_YAMLS);\
+	$(eval YAMLS=$(shell sh -c "find schedule/ test_data/ -regex '.*\.ya?ml$$'"))
```

Comment:
It's faster to use `git ls-files` to only list all version controlled files but does a different thing than in before. Likely @rwx788 can comment best on this

Reply:
I guess git commands are always faster than generic commands. Nevertheless, what concerns me more is that now we run this check for all yaml files. I would suggest that if we want to check all, we do it only once when PR is merged or even just regularly, and for open PRs we check only modified and added files.
Am I right that main motivation here is not to miss badly formatted yamls, like we did for those files you have fixed?

Reply:
yes - and  the existing PR checks don't seem to cover them - maybe github is simulating a merge in the PR github action.
Checking all YAML files takes only 9 seconds, so I'm not sure if it would be worth the effort to change this.

Reply:
I believe we will get more yaml files in future, so it might take longer. However, we can also do it as a next step (could you please file a ticket for that?).
I would still propose `git ls-files` command here.

Reply:
@rwx788 I updated the commit tu use `git ls-files`